### ISSUE TITLE
/vehicle /tpa issues & questions

### ISSUE THREAD

**Issue by qeary** [uEssentials/uEssentials#670]
Hello! I would like to ask you if you can help me..
1. /vehicle or /v is not really working smooth for me.. if i turn it off from UESSENTIALS then "command not found" even the rocketmod has it by itself. I wish to use the basic /v
2. Please answer the 1. too but if maybe there is other chance, to report you my problem is: if someone does /v but dont spawn any vehicle it will still get the cooldown. Maybe i shall add cooldown for it by the uessentials? If i shall, how?
3. Here is my third question, connect to the second one. Right now im using your /tpa but i added cooldowns for it from the permissions config file, can i add cooldown through uessentials? (i see something that shall do it for me, its called "command-options.json" but i dont understand how to use it :/
4. Is there any way to make cooldown for sending tpa-s but NO COOLDOWN for accepting them?
5. Bug report: I can send anytimes /tpa NAME, but i wish to use it only once and then it shall say "You already sent TPA to "NAME"
6. Back to vehicles.. if i keep using your /v, can i remove the stuff that allow players to spawn vehicles for all players? So i want them to only /v ID to themselves.
7. By this, i didnt find any way to change it's message, i checked translations.. :/ "Use /vehicle [vehicle] or [player|* = all] [vehicle]" Well i dont want them to even know about this feature, only "/vehicle ID"

SORRY if i was explaining too long, i hope you can understand everything and thanks for the answer!

**Comment by leonardosnt** [comment]
1. How are you 'turning it off'?
2. You can do that in the `command_options.json`. Ex:
```json
"vehicle": {
   "Cooldown": 20
 }
``` 
3. Do the same thing of the 2°
4. No
5. Please, create a separated issue for that.
6. Actually, the player needs the permission: `essentials.command.vehicle.other` to be able to give a vehicle to another player.
7. That's not possible.

**Comment by qeary** [comment]
1. Disabledcommands: "vehicle",
2. This cooldown will stay after server restart?
2 ----- This cooldown will stay after /rocket reload?
2 ---------- This thing will make it not goes to be on cooldown if they only do /v to see the command usage?
3. Thanks :)
4. Thats sad but then how can i avoid spamming?:/
5. Separated issue for that = shall i open a NEW issue only for this thing?
6. Okey so are you sure if i do only essentials.command.vehicle they wont be able to spawn vehicles for other players? I ask this if you are sure cuz the command description looks like this:
"""""""""""""""""""""""""""""""""""
Name: vehicle
Usage: /vehicle [vehicle] or [player|* = all] [vehicle]
Aliases:
- v
Permissions:
- essentials.command.vehicle
""""""""""""""""""""""""""""""""""""
By this.. if you can add ALL the usable permissions for this command?
Or maybe this permission will allow EVERY permission for vehicles? So ".other".. etc? If yes, then what to add them if i want only to let them only DO /vehicle (or /v) ID ?

7. Is it really not possible? I can set many many translations but not the vehicle stuff? That would be a real problem.


(Please remmeber to answer for the 1. question too cuz u just asked "how are you turning it off" and didnt answered the question there.)

+++++++++++
Thanks for these point by point answers, i really love it, its really "understandable"!

**Comment by leonardosnt** [comment]
1. Weird, that should have worked...  Put your `config.json` here, please.
2. Yes.
  2.1. Yes
  2.2. Yes. The cooldown will be only applied if the command was successfully executed.  
3. No problem.
4. Just add a cooldown :\\
5. Yes
6. Yes. The usage is 'global', it is not one for each permission. (Idk if this 'sentence' is right because my English is not that good (isn't my first language))
7. Wait... Sorry 😆. Yes, you can do that in the `command_options.json`. E.g
```json
"vehicle": {
  "Usage": "Usage here",
  "Cooldown": 20
}
```

**Comment by qeary** [comment]
1. Uhh, you missunderstood me! /vehicle is working, but i wanna know how can i use the rocketmod basic /vehicle. If i do config.json disabledcommands > "vehicle", so i turn off from uessentials, it wont be usable, wont be turned back to the basic /vehicle from the rocketmod. Thats what i want, to use the basic one.
4. Are you sure you can't do it? (The single TPA plugin works like this, i still wish to use your plugin and i can add cooldown, just asking ^^) (Or even you just don't want to do it like this? Im ok with that.) You still made a really good job with that plugin!
6. Wait. what, usage is global? so umm i dont understand. You said to let them add vehicles for other players i need to do "essentials.command.vehicle.other" but now you said i dont need to do this, its enough to do "essentials.command.vehicle"
So to be clear: I want them to NOT BE ABLE to add vehicles to other players only to themself!

**Comment by qeary** [comment]
(3. = Solved, i checked it on lan server, know the way to work! dont asnwer for 3. question x) )

**Comment by leonardosnt** [comment]
1. If you put the command in the `disabledcommands` list you should be able to use the RocketMod `/vehicle`. Did you tried it?
2. The current implementation of the `command_options` doesn't allow to add `cooldown` to a _sub command_ (like /tpa **accept**). Unfortunately, I don't have free time right now to implement this _feature_.

**Comment by qeary** [comment]
1. Yes i tried it, also.. i tried it on LAN server too and there it worked but i stay using your vehicle, cuz its good for now, i understand everything with that so its like "nevermind" anymore :)
2. Alright, i wish you to have some time for it, but until this time, i just use the cooldown for all :D
Its okey.
Thanks for every answers! <3
